core/issuers.chainmail

```diff
@@ -1,56 +1,183 @@
+ * transfer rights. The only use of a currency is to spend it. Some goods
+ * distinguish use rights from transfer rights, like a membership card; you can
+ * exercise the rights without giving them up.
+ */
+interface Purse extends (PurseOrPayment) {
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+  getXferBalance() -> (Amount);
+  getUseBalance() -> (Amount);
+  deposit(amount:Amount, srcPayment:PurseOrPayment) -> (Amount);
+  depositAll(srcPayment:PurseOrPayment) -> (Amount);
+  withdraw(amount:Amount, name:String) -> (Payment);
+  withdrawAll(name:String) -> (Payment);
 }
-makeNatAssay(label ::Label) ::Assay;
-makeMetaSingleAssayMaker(
-  baseLabelToAssayFn ::(Label -> Assay)) ::(Label -> Assay);
-
-class Issuer[Assay] {
-  type Amount = Assay.Amount;
-  getLabel() ::{ issuer ::Issuer, description };
-  getAssay() ::Assay;
-  makeEmptyPurse(name ?String) ::Purse;
-
-  class Mint {
-    getIssuer() Issuer;
-    mint(initialBalance ?Amount, name ?String) ::Purse;
-  }
-  class Payment {
-    getIssuer() ::Issuer;
-    getXferBalance() ::Amount;
-  }
-  class Purse {
-    getIssuer() ::Issuer;
-    getXferBalance() ::Amount;
-    getUseBalance() ::Amount;
-    deposit(amount ?Amount, srcPaymentP ?reveal[Promise]) ::Amount;
-    withdraw(amount ?Amount, name ?String) ::Purse;
-  }
+
+/**
+ * Payments hold verified amounts of certain rights issued by Mints. Amounts
+ * from payments can be deposited in purses, but otherwise, the entire amount is
+ * available when the payment is transferred. A payment's balance can only fall,
+ * through the action of deposit() or burn().
+ * 
+ * Payments can be converted to Purses by calling
+ * payment.getIssuer().makeEmptyPurse().depositAll(payment);
+ */
+interface Payment extends (PurseOrPayment) {
+  getIssuer() -> (Issuer);
+  getXferBalance() -> (Amount);
+}
+
+/**
+ * PurseOrPayment is the parent type of Purse and Payment. Most interfaces only
+ * accept one of the sub-types to reduce the incidence of the common mistake of
+ * giving away a purse when you meant to hold on to the purse and give away part
+ * of the contents. deposit() and depositAll() are the two exceptions, for which
+ * the caller explicitly says to give up all contained rights, or provides an
+ * explicit amount to transfer.
+ */
+interface PurseOrPayment {
+  getIssuer() -> (Issuer);
+  getXferBalance() -> (Amount);
 }
-makeMint(description, makeAssay ::(Label -> Assay)) ::Issuer[Assay].Mint;
-
-class Peg[RemoteIssuer, LocalIssuer] {
-  getLocalIssuer() ::LocalIssuer;
-  getRemoteIssuer() ::reveal<RemoteIssuer>;
-  retain(remoteAmount ?reveal<RemoteIssuer.Amount>,
-         remotePaymentP ?reveal<RemoteIssuer.Payment>,
-         name ?String) ::LocalIssuer.Payment;
-  redeem(localAmount ?LocalIssuer.Amount,
-         localPayment ?LocalIssuer.Payment,
-         name ?String) ::reveal<RemoteIssuer.Payment>;
+
+/** MintController records the amounts contained by purses. */
```

Comment:
MintController somehow seems more internal. Can we avoid exposing this type in issuers.chainmail ?

Reply:
Since this is about to change so completely and is not exposed for use anyway, I think you can just omit it. This part of the API on master can remain undocumented until Kate's PR.